core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -85,181 +102,250 @@ protected function configure(): void
-        if ($concurrent < 1) {
-            throw new InvalidArgumentException(sprintf('Concurrent value "%s" is invalid.', $concurrent));
+        if (!$output instanceof ConsoleOutputInterface) {
+            throw new \InvalidArgumentException(sprintf('Output must be an instance of "%s"', ConsoleOutputInterface::class));
         }
 
-        $io = new SymfonyStyle($input, $output);
-
-        if ($concurrent > 1) {
-            return $this->executeConcurrent($concurrent, $io);
+        if (null !== $image = $input->getOption('image')) {
+            return $this->resizeImage($image);
         }
 
-        $startTime = microtime(true);
-
-        $maxDuration = 0;
-        $lastDuration = 0;
-
-        foreach ($this->storage->listPaths() as $path) {
-            $sleep = (1 - $throttle) / $throttle * $lastDuration;
+        $timeLimit = (float) $input->getOption('time-limit');
+        $throttle = (float) $input->getOption('throttle');
+        $concurrent = (float) $input->getOption('concurrent');
 
-            if ($timeLimit && microtime(true) - $startTime + $maxDuration + $sleep > $timeLimit) {
-                $io->writeln("\n".'Time limit of '.$timeLimit.' seconds reached.');
+        if (1.0 !== $throttle) {
+            @trigger_error('Using the throttle option is deprecated and will no longer work in Contao 5.0. Use the concurrent option instead.', E_USER_DEPRECATED);
+            $this->io->warning('Using the throttle option is deprecated, use the concurrent option instead.');
 
-                return 0;
+            if ($throttle < 0.001 || $throttle > 1) {
+                throw new InvalidArgumentException(sprintf('Throttle value "%s" is invalid.', $throttle));
             }
 
-            usleep((int) ($sleep * 1000000));
+            $concurrent *= $throttle;
+        }
+
+        if ($timeLimit < 0) {
+            throw new InvalidArgumentException(sprintf('Time-limit value "%s" is invalid.', $timeLimit));
+        }
 
-            $maxDuration = max($maxDuration, $lastDuration = $this->resizeImage($path, $io));
+        if ($concurrent <= 0) {
+            throw new InvalidArgumentException(sprintf('Concurrent value "%s" is invalid.', $concurrent));
         }
 
-        $io->writeln("\n".'All images resized.');
+        $this->tableOutput = $output->section();
 
-        return 0;
+        $this->table = new Table($this->tableOutput);
+        $this->table->setHeaders(['Process ID', 'Count', 'Image']);
+        $this->table->setColumnWidth(2, $this->terminalWidth - 25);
+        $this->table->setColumnMaxWidth(2, $this->terminalWidth - 25);
+
+        return $this->resizeImages($timeLimit, $concurrent);
     }
 
-    private function resizeImage(string $path, SymfonyStyle $io): float
+    private function resizeImage(string $path, bool $quiet = false): int
     {
-        $startTime = microtime(true);
-
-        if ($this->filesystem->exists($path)) {
+        if ($this->filesystem->exists($this->targetDir.'/'.$path)) {
```

Comment:
This line + below should probably also be converted into a `Path::join()` + variable when merging upstream. :slightly_smiling_face: 

/cc @leofeyer

Reply:
See #2041.